### PR TITLE
test: thumbnail component test

### PR DESCRIPTION
**Pull request by AntonOlkhovskyi**
#### Please check whether the PR fulfills the following requirements

- [ x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [ x] tests for the changes that have been done
- [x ] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x ] update `README.md`
- [ x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [ x] Documentation Examples
- [ x] Stackblitz works for all examples

### CODE REVIEW

File: e2e/wdio/driver/wdio.ts [hunk 1]
```diff
@@ -267,3 +267,13 @@ export function mouseButtonUp(button: 0 | 1 | 2 = 0): void {
...
     $$(selector)[index].nextElement().click();
 }
+
+export function getElementLocation(selector: string, index?: number):  WebdriverIO.LocationReturn;
+export function getElementLocation(selector: string, index: number, prop: 'x' | 'y'): number;
+export function getElementLocation(selector: string, index: number = 0, prop?: 'x' | 'y'): WebdriverIO.LocationReturn | number {
+    return prop ? $$(selector)[index].getLocation(prop) : $$(selector)[index].getLocation();
+}
+
+export function getPerrentElementCSSPropetry(selector: string, prop: string, index: number): string {
```
Comment (Seamoo13) [inline e2e/wdio/driver/wdio.ts:277]: spelling error. Parent, not Perrent

File: e2e/wdio/platform/pages/thumbnail.po.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    mainImage = '.fdp-media-container img';
+    verticalGalleryImages = 'fdp-thumbnail-basic-example fd-avatar';
+    horizontalGalleryImages = 'fdp-thumbnail-horizontal-example fd-avatar';
+    verticalGalleryVideo = 'fdp-thumbnail-video-media-example fd-avatar';
+
+    open(): void {
+        super.open(this.url);
+        waitElementToBePresentInDOM(this.root, 30000);
+        waitForElDisplayed(this.root, 30000);
```
Comment (Seamoo13) [inline e2e/wdio/platform/pages/thumbnail.po.ts:17]: let's use Nikita's solution by adding waitForPresent(this.mainImg); to the open() method here and after the refresh in the afterEach hook in the spec

File: e2e/wdio/platform/tests/thumbnail.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+describe('Thumbnail field', function() {
+    const thumbnailPage: ThumbnailPo = new ThumbnailPo();
+
+    beforeAll(() => {
+        thumbnailPage.open();
+    });
+
+    afterEach(() => {
+        refreshPage();
```
Comment (Seamoo13) [inline e2e/wdio/platform/tests/thumbnail.e2e-spec.ts:24]: please add waitForPresent(thumbnailPage.mainImg); after refreshPage();

**Review by Seamoo13 — CHANGES_REQUESTED** [review]

File: e2e/wdio/driver/wdio.ts (hunk 1 above)
Comment (AntonOlkhovskyi) [reply e2e/wdio/driver/wdio.ts:277]: Done.

File: e2e/wdio/platform/pages/thumbnail.po.ts (hunk 2 above)
Comment (AntonOlkhovskyi) [reply e2e/wdio/platform/pages/thumbnail.po.ts:17]: renamed waitElementToBePresentInDOM to waitForPresent

File: e2e/wdio/platform/tests/thumbnail.e2e-spec.ts (hunk 3 above)
Comment (AntonOlkhovskyi) [reply e2e/wdio/platform/tests/thumbnail.e2e-spec.ts:24]: Done

File: e2e/wdio/platform/pages/thumbnail.po.ts (hunk 2 above)
Comment (nikvalor) [reply e2e/wdio/platform/pages/thumbnail.po.ts:17]: open(): void {
        super.open(this.url);
        waitForPresent(this.mainImage, 30000);
        waitForElDisplayed(this.mainImage, 30000);

File: e2e/wdio/platform/tests/input.e2e-spec.ts [hunk 4]
```diff
@@ -40,13 +40,13 @@ describe('Input should ', function() {
...
-        waitElementToBePresentInDOM(inputPage.defaultInput);
+        waitForPresent(inputPage.defaultInput);
         waitForElDisplayed(inputPage.defaultInput);
         expect(doesItExist(inputPage.autocompleteInputLabel)).toBe(false);
     });
 
     it('be able to type something with keyboard', () => {
-        waitElementToBePresentInDOM(inputPage.defaultInput);
+        waitForPresent(inputPage.defaultInput);
         waitForElDisplayed(inputPage.defaultInput);
```
Comment (valorkin) [inline e2e/wdio/platform/tests/input.e2e-spec.ts:50]: if `waitForElDisplayed` used mostly with `waitForPresent`, then maybe `waitForPresent` should include `waitForPresent` inside of it? to reduce boiler plate?

File: e2e/wdio/platform/tests/input.e2e-spec.ts [hunk 5]
```diff
@@ -128,7 +128,7 @@ describe('Input should ', function() {
 
     it('should have error border color', () => {
         if (!browserIsIEorSafari()) {
```
Comment (valorkin) [inline e2e/wdio/platform/tests/input.e2e-spec.ts:130]: invert and reduce nesting

File: wdio.conf.js [hunk 6]
```diff
@@ -49,23 +49,23 @@ exports.config = {
...
-            browserName: 'internet explorer',
-            browserVersion: 'latest',
-            platformName: 'Windows 10',
-            'sauce:options': {
-                screenResolution: '1920x1080',
-                name: 'e2e-win-internet-explorer ' + process.env.TRAVIS_BUILD_ID + ' ' + process.env.TRAVIS_PULL_REQUEST_BRANCH,
-                requireWindowFocus: true,
-            }
-        },
+        // {
```
Comment (valorkin) [inline wdio.conf.js:59]: ?

**Review by valorkin — CHANGES_REQUESTED** [review]
there are code style related suggestions, mentioned in one place but expeced to be applied to all relevant places

File: e2e/wdio/platform/tests/thumbnail.e2e-spec.ts [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+        waitForElDisplayed(thumbnailPage.mainImage);
+        expect(isElementDisplayed(thumbnailPage.mainImage)).toBeTrue();
+        expect(isElementDisplayed(thumbnailPage.mainImage, 1)).toBeTrue();
+        expect(isElementDisplayed(thumbnailPage.mainImage, 2)).toBeTrue();
+    });
+
+    it('should be able to display images vertical', () => {
+        scrollIntoView(thumbnailPage.verticalGalleryImages);
+        const arrLength = getElementArrayLength(thumbnailPage.verticalGalleryImages);
+        for (let i = 0; arrLength > i; i++) {
```
[the remark this answers is not in the document]
Comment (AntonOlkhovskyi) [reply e2e/wdio/platform/tests/thumbnail.e2e-spec.ts:39]: Done.

File: e2e/wdio/driver/wdio.ts [hunk 8]
```diff
@@ -267,3 +263,13 @@ export function mouseButtonUp(button: 0 | 1 | 2 = 0): void {
 export function  clickNextElement(selector: string, index: number = 0): void {
     $$(selector)[index].nextElement().click();
 }
+
+export function getElementLocation(selector: string, index?: number):  WebdriverIO.LocationReturn;
+export function getElementLocation(selector: string, index: number, prop: 'x' | 'y'): number;
+export function getElementLocation(selector: string, index: number = 0, prop?: 'x' | 'y'): WebdriverIO.LocationReturn | number {
+    return prop ? $$(selector)[index].getLocation(prop) : $$(selector)[index].getLocation();
```
[the remark this answers is not in the document]
Comment (AntonOlkhovskyi) [reply e2e/wdio/driver/wdio.ts:270]: changed to return $$(selector)[index].getLocation(prop || void 0);

File: e2e/wdio/platform/tests/input.e2e-spec.ts (hunk 4 above)
Comment (AntonOlkhovskyi) [reply e2e/wdio/platform/tests/input.e2e-spec.ts:50]: fixed

File: e2e/wdio/platform/tests/input.e2e-spec.ts (hunk 5 above)
Comment (AntonOlkhovskyi) [reply e2e/wdio/platform/tests/input.e2e-spec.ts:130]: done

File: wdio.conf.js (hunk 6 above)
Comment (AntonOlkhovskyi) [reply wdio.conf.js:59]: Skipping IE for now because of random failing tests on the CI.

**Review by nikvalor — APPROVED** [review]

File: e2e/wdio/platform/pages/thumbnail.po.ts [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+    mainImage = '.fdp-media-container img';
+    mainVideo = '.fdp-media-container video';
+    verticalGalleryImages = 'fdp-thumbnail-basic-example fd-avatar';
+    horizontalGalleryImages = 'fdp-thumbnail-horizontal-example fd-avatar';
+    verticalGalleryVideo = 'fdp-thumbnail-video-media-example fd-avatar';
+
+    open(): void {
+        super.open(this.url);
+        waitForElDisplayed(this.mainImage);
```
Comment (Seamoo13) [inline e2e/wdio/platform/pages/thumbnail.po.ts:17]: waitForPresent?

Comment (AntonOlkhovskyi) [reply e2e/wdio/platform/pages/thumbnail.po.ts:17]: waitForElDisplayed is now includes waitForPresent according to this comment 
> if `waitForElDisplayed` used mostly with `waitForPresent`, then maybe `waitForPresent` should include `waitForPresent` inside of it? to reduce boiler plate?

Comment (Seamoo13) [reply e2e/wdio/platform/pages/thumbnail.po.ts:17]: ok

**Review by valorkin — APPROVED** [review]
lgtm with one comment in DM

**Review by Seamoo13 — APPROVED** [review]

**Review by EvilAlexei — APPROVED** [review]

**Pull request merged**